net-p2p/bitcoin-qt/bitcoin-qt-0.16.0.ebuild

```diff
@@ -0,0 +1,232 @@
+"
+COMMON_DEPEND="
+	!libressl? ( dev-libs/openssl:0=[-bindist] )
+	libressl? ( dev-libs/libressl:0= )
+	libevent? ( dev-libs/libevent:= )
+	>=dev-libs/libsecp256k1-0.0.0_pre20151118:=[recovery]
+	dev-libs/univalue:=
+	>=dev-libs/boost-1.52.0:=[threads(+)]
+	upnp? ( >=net-libs/miniupnpc-1.9.20150916:= )
+	wallet? ( sys-libs/db:$(db_ver_to_slot "${DB_VER}")=[cxx] )
+	zeromq? ( net-libs/zeromq:= )
+	virtual/bitcoin-leveldb
+	bip70? ( dev-libs/protobuf:= )
+	qrcode? (
+		media-gfx/qrencode:=
+	)
+	dev-qt/qtcore:5=
+	dev-qt/qtgui:5=
+	dev-qt/qtnetwork:5=
+	dev-qt/qtwidgets:5=
+	dbus? (
+		dev-qt/qtdbus:5=
+	)
+"
+DEPEND="${COMMON_DEPEND}
+	dev-qt/linguist-tools:5
+	knots? (
+		gnome-base/librsvg
+		media-gfx/imagemagick[png]
+	)
+"
+RDEPEND="${COMMON_DEPEND}"
+
+declare -A LANG2USE USE2LANGS
+bitcoin_langs_prep() {
+	local lang l10n
+	for lang in ${LANGS}; do
+		l10n="${lang/:*/}"
+		l10n="${l10n/[@_]/-}"
+		lang="${lang/*:/}"
+		LANG2USE["${lang}"]="${l10n}"
+		USE2LANGS["${l10n}"]+=" ${lang}"
+	done
+}
+bitcoin_langs_prep
+
+bitcoin_lang2use() {
+	local l
+	for l; do
+		echo l10n_${LANG2USE["${l}"]}
+	done
+}
+
+IUSE+=" $(bitcoin_lang2use ${!LANG2USE[@]})"
+
+DOCS=( doc/bips.md doc/files.md doc/release-notes.md )
+
+S="${WORKDIR}/bitcoin-${BITCOINCORE_COMMITHASH}"
+
+pkg_pretend() {
+	if use knots; then
+		einfo "You are building ${PN} from Bitcoin Knots."
+		einfo "For more information, see https://bitcoinknots.org/files/0.16.x/${KNOTS_PV}/${KNOTS_P}.desc.html"
+	else
+		einfo "You are building ${PN} from Bitcoin Core."
+		einfo "For more information, see https://bitcoincore.org/en/2017/11/11/release-${PV}/"
+	fi
+	if use bitcoin_policy_rbf; then
+		einfo "Replace By Fee policy is enabled: Your node will preferentially mine and relay transactions paying the highest fee, regardless of receive order."
+	else
+		einfo "Replace By Fee policy is disabled: Your node will only accept the first transaction seen consuming a conflicting input, regardless of fee offered by later ones."
+	fi
+}
+
+src_prepare() {
+	sed -i 's/^\(complete -F _bitcoind \)bitcoind \(bitcoin-qt\)$/\1\2/' contrib/bitcoind.bash-completion || die
+
+	local KNOTS_PATCHDIR="${WORKDIR}/${KNOTS_P}.patches/"
+
+	eapply "${KNOTS_PATCHDIR}/${KNOTS_P}.syslibs.patch"
+
+	if use knots; then
+		eapply "${KNOTS_PATCHDIR}/${KNOTS_P}.f.patch"
+		eapply "${KNOTS_PATCHDIR}/${KNOTS_P}.branding.patch"
+		eapply "${KNOTS_PATCHDIR}/${KNOTS_P}.ts.patch"
+		eapply "${FILESDIR}/${P}-fix_mempoolstats.patch"
+	fi
+
+	eapply_user
+
+	if ! use bitcoin_policy_rbf; then
+		sed -i 's/\(DEFAULT_ENABLE_REPLACEMENT = \)true/\1false/' src/validation.h || die
+	fi
+
+	echo '#!/bin/true' >share/genbuild.sh || die
+	mkdir -p src/obj || die
+	echo "#define BUILD_SUFFIX gentoo${PVR#${PV}}" >src/obj/build.h || die
+
+	sed -i 's/^\(Icon=\).*$/\1bitcoin-qt/;s/^\(Categories=.*\)$/\1P2P;Network;Qt;/' contrib/debian/bitcoin-qt.desktop || die
+
+	local filt= yeslang= nolang= lan ts x
```

Comment:
This is not a valid use of L10N flags. Translations are to be installed unconditionally, so please remove all the relevant logic. This will also make the ebuild much simpler.

Reply:
In that case, what do `L10N` flags exist for?

Reply:
Language packs, big and/or separately fetched stuff.